File: Euclid/Game/Item/ItemDefinition.cs
using Euclid.Storage.Database.Data;

namespace Euclid.Game
{
    public class ItemDefinition
    {
        #region Properties

        public ItemDefinitionData Data { get; }

        #endregion

        #region Constructors

        public ItemDefinition(ItemDefinitionData data)
        {
            Data = data;
        }

        #endregion

        #region Public methods

        /// <summary>
        /// Get top height, but any top height below 0 will be returned as 0.
        /// </summary>
        /// <returns></returns>
        public double GetPositiveTopHeight()
        {
            if (Data.Height < 0)
            {
                return 0;
            }

            return Data.Height;
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/game/item/ItemDefinition.java
package euclid.game.item;

import euclid.storage.database.data.ItemDefinitionData;

public class ItemDefinition {

    private final ItemDefinitionData data;

    public ItemDefinition(ItemDefinitionData data) {
        this.data = data;
    }

    public ItemDefinitionData getData() {
        return data;
    }

    public double getPositiveTopHeight() {
        if (data.getHeight() < 0) {
            return 0;
        }
        return data.getHeight();
    }
}
